lib/net.js

```diff
@@ -1137,11 +1137,27 @@ function Server(options, connectionListener) {
     configurable: true, enumerable: false
   });
 
+  Object.defineProperty(this, '_handle', {
+    set: function(val) {
+      if (val === null) {
+        this._listenHasBeenCalled = false;
+      }
+      this.__handle = val;
```

Comment:
To really hide it from users you can use a internal symbol:

```
const handleSymbol = Symbol('handle');
...
this[handleSymbol] = val;
...
return this[handleSymbol];
```

The same can be done to `_listenHasBeenCalled` too if you want to hide it. Old properties probably can't be hidden this way in case people are already using it..but new properties can ;)